data/src/lib.rs

```diff
@@ -0,0 +1,82 @@
+//! The single source of truth for all data paths.
+//!
+//! Run `cargo make gen-assets` before using this!
+
+pub mod bench;
+pub mod example;
+pub mod ltr;
+pub mod qambert;
+pub mod smbert;
+pub mod test;
+
+use std::{
+    collections::HashMap,
+    env::{current_dir, var_os},
+    fs::File,
+    io::{BufReader, Error, ErrorKind, Result},
+    path::{Path, PathBuf},
+};
+
+use serde::Deserialize;
+use serde_json::{from_reader, from_value, Value};
+
+/// Resolves the path to the requested data.
+fn resolve_path(path: &[impl AsRef<Path>]) -> Result<PathBuf> {
+    let manifest = var_os("CARGO_MANIFEST_DIR")
+        .ok_or_else(|| Error::new(ErrorKind::NotFound, "missing CARGO_MANIFEST_DIR"))?;
+    let manifest = PathBuf::from(manifest).canonicalize()?;
+    let current = current_dir()?.canonicalize()?;
+
+    let workspace = if current == manifest {
+        current
+            .parent()
+            .ok_or_else(|| Error::new(ErrorKind::NotFound, "missing workspace"))?
+            .to_path_buf()
+    } else if Some(current.as_path()) == manifest.parent() {
+        current
+    } else {
+        return Err(Error::new(ErrorKind::NotFound, "missing workspace"));
+    };
+
+    path.iter()
+        .fold(workspace, |path, component| path.join(component))
+        .canonicalize()
+}
+
+#[derive(Deserialize)]
+struct Asset {
+    name: String,
+    path: String,
+}
+
+#[derive(Deserialize)]
+struct Assets {
+    assets: Vec<Value>,
```

Comment:
unfortunately this doesn't work because the maps in the array are no homogeneuous, for example the maps related to wasm will crash the deserialization. this is why i had to go the `Value` way.

Reply:
I think this is more like a bug of the templates.

The relevant part is e.g.:
```handlebars
      "name": "wasmModule",
      {{ if ( $wasm_module_path | file.Exists ) -}}
      "suffix": "{{ $wasm_module_suffix }}",
      "checksum": "{{ $wasm_module_path | file.Read | crypto.SHA256 }}",
      "path": "{{ $wasm_module_path }}"
      {{ else -}}
      "suffix": "",
      "checksum": ""
      {{ end -}}
 ```
 
 but if we anyway set `"checkum"` and `"suffix"` to `""` couldn't we just do the same for `"path"`?

Reply:
> but if we anyway set "checkum" and "suffix" to "" couldn't we just do the same for "path"?
yes that shouldn't be an issue

Reply:
i agree, i'd suggest to fix this in a follow up pr.

Reply:
if we can simplify it to only have one of the two approaches, it probably makes sense.

Reply:
What is the benefit of removing `path`? Is the value of suffix then `<data_dir>/<suffix>`?

Reply:
I thought a bit more about the idea of removing `path`. The only issue I see here is that the wasm artifacts are not stored in the `data_dir` so `<data_dir>/<suffix>` wouldn't work in the CI. However we can remove `path` in the [static json file](https://github.com/xaynetwork/xayn_ai/pull/156/files#diff-0bcfa4d5d47b43679a40f5a7bfd737f2de808280dcdc3fc02203c5127077632e) so that we only have `suffix`. 
wdyt?